c10/util/Half-math.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <c10/util/Half.h>
+#include <c10/util/math_compat.h>
+
+namespace std {
+
+/// Used by vec256<c10::Half>::map
+inline c10::Half acos(c10::Half a) { return std::acos(float(a));}
```

Comment:
I'm not sure what the history of half math is. @ngimel, do you know?

Reply:
No, I don't. I once wondered how math overloads work, but didn't get a full picture, I wonder why it's necessary, shouldn't there be an implicit Half->float conversion (that's defined)? I thought that's how std math functions work in cuda kernels.

Reply:
I assumed `BFloat-math.h` was an accepted pattern, but if that's not true then I've just removed it and focused on `copysign`.